### ISSUE TITLE
update broke content control with PHP 7.2.34

### ISSUE THREAD

**Issue by twoprops** [code-atlantic/content-control#76]
One of my sites was happily using Content Control until 2.0, when an automatic update brought the whole site down. I had to manually deactivate the plugin from the command line. When I try to re-activate it, I get an error message: "Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 7.4.0".  Note that the Content Control plugin page says "PHP Version: 5.6 or higher". I'm running 7.2.34 and won't be able to update for another ten days (it's a long story). Is there any way to get my site working again short of updating PHP? Is there any way to get advance warning when you're going to make critical changes to dependencies? Thanks!

**Comment by danieliser** [comment]
@twoprops - I'm assuming you must be installing using composer setup of some sort, not via wordpress.org updates?

Sorry I honestly didn't realize people were using that and haven't even tested it. I have a feeling that our late building of assets for release (not committed) is the culprit.

If so we are already investigating a process change when we make releases so that we will make a new branch, build the plugin release and delete everything else, commit the changes, tag the branch as the release. That way the source zips on Github releases would be the same code as deployed to wordpress.org, which isn't the case now.

You can definitely revert in composer to a specific version  `v1.1.10`: https://github.com/code-atlantic/content-control/releases/tag/v1.1.10

I think  it would be something like `composer require package:1.1.10`, or just set the composer.json package to `1.1.10`.

Let me know if that is the case.

**Comment by twoprops** [comment]
I have never, to my knowledge, used composer. I had to look up what it is (I don't really do anything with PHP outside of WordPress). That said, I vaguely remember there was some glitch installing WP for this project and it's possible I inadvertently did something that installed composer. A search for `composer` on the system reveals these files:
````
/srv/www/wordpress/wp-includes/sodium_compat/composer.json
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/platform_check.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/installed.json
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/autoload_psr4.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/autoload_real.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/ClassLoader.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/autoload_classmap.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/autoload_namespaces.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/installed.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/autoload_static.php
/srv/www/wordpress/wp-content/plugins/content-control/vendor/composer/InstalledVersions.php
/snap/core/16091/usr/lib/python3/dist-packages/yaml/__pycache__/composer.cpython-35.pyc
/snap/core/16091/usr/lib/python3/dist-packages/yaml/composer.py
/snap/core/15925/usr/lib/python3/dist-packages/yaml/__pycache__/composer.cpython-35.pyc
/snap/core/15925/usr/lib/python3/dist-packages/yaml/composer.py
/usr/lib/python3/dist-packages/yaml/composer.py
/usr/lib/python3/dist-packages/yaml/__pycache__/composer.cpython-35.pyc
/usr/share/man/man1/dh_phpcomposer.1.gz
/usr/share/sosreport/sos/plugins/composer.py
/usr/share/sosreport/sos/plugins/__pycache__/composer.cpython-35.pyc
/usr/share/perl5/Debian/Debhelper/Sequence/phpcomposer.pm
/usr/share/php/pkgtools/phpcomposer
/usr/share/php/pkgtools/phpcomposer/source.php
/usr/share/php/pkgtools/phpcomposer/command.php
/usr/bin/dh_phpcomposer
````
My recollection, however, is that I did a stock install of WordPress from the wordpress.org site and installed content control in the usual way from the WordPress store.

Sorry to not be more knowledgeable about this, but I wouldn't have the first clue how to revert to a previous version of a plugin given just a github link.

**Comment by twoprops** [comment]
Your response led me to investigate composer more fully. I'm still not sure how it got there, but I remember being rushed/distracted when doing the install so I probably just tried a suggestion that installed it. Once you led me to the source of the problem, I was able to resolve it by temporarily disabling the version check until I can Do The Right Thing (update PHP) later this week. Thanks, and sorry for the bother.

**Comment by danieliser** [comment]
@twoprops - You got it working then? Were you installing the plugin from composer after all?

Looking over the file search, that is normal looking. 

https://plugins.trac.wordpress.org/browser/content-control/#trunk/vendor

**Comment by danieliser** [comment]
For the record, our dev-dependencies when working on the plugin require PHP >7.4. That is in our composer files, not sure how that would show up in your WP dash though.

**Comment by twoprops** [comment]
Yes, I got it working. Yes, it does appear that composer got in the loop somehow but I have absolutely no recollection of installing it. I'd never even heard of it, which is why the error message didn't trigger a more appropriate reaction. I suspect I ran into a problem during my WordPress install and found a workaround on Stack Overflow or some such and -- probably in an  after-midnight frenzy -- blindly followed a recipe that included composer. The problem was in `...wordpress/wp-content/plugins/content-control/vendor/composer/platform_check.php`. I simply commented out the line `if (!(PHP_VERSION_ID >=  ...` and everything seems to be working fine. I need to update PHP for a number of reasons but I have a dependency that requires the old version and I'm currently overseas. I can fix that problem when I'm back on site later this week, and I'll be sure to revert the change once PHP is up to date.

Thanks for your help, particularly as it appears that the problem wasn't with content control but with my install. More importantly, thanks for making content control. I only use a fraction of its capabilities, but it turned out to be hugely important for my application. It appears to be a well-written, concise, and elegant solution to several shortcomings I've found with WordPress.
